src/plugins/intel_gna/src/gna_plugin_config.cpp

```diff
@@ -46,7 +46,56 @@ static const caseless_unordered_map<std::string, std::pair<Gna2AccelerationMode,
 };
 OPENVINO_SUPPRESS_DEPRECATED_END
 
+/**
+ * @brief Parse string into value without throwing
+ * @param value A parsed value
+ * @param def_value A value to return in case of exception
+ * @return parsed value, default value otherwise
+ * @ingroup ov_runtime_gna_prop_cpp_api
+ */
+template <class T>
+static T FromString(const std::string& value, T def_value) {
+    T result = def_value;
+    try {
+        result = ov::util::from_string<T>(value);
+    } catch (...) {
+    }
+    return result;
+};
+
+/**
+ * @brief Set the gnaPrecision value having both needed input values
+ * @param inference_precision_value Value passed for inference_precision property
+ * @param execution_mode_value Value passed for execution_mode property
+ * @ingroup ov_runtime_gna_prop_cpp_api
+ */
+void Config::UpdateGNAPrecision(const std::string& inference_precision_value, const std::string& execution_mode_value) {
+    if (!execution_mode_value.empty() && inference_precision_value.empty()) {
+        execution_mode = FromString<ov::hint::ExecutionMode>(execution_mode_value, ov::hint::ExecutionMode::UNDEFINED);
+        if ((execution_mode != ov::hint::ExecutionMode::ACCURACY) &&
+            (execution_mode != ov::hint::ExecutionMode::PERFORMANCE)) {
+            THROW_GNA_EXCEPTION << "Unsupported execution mode, should be accuracy or performance, but was: "
+                                << execution_mode_value;
+        }
+        gnaPrecision = (execution_mode == ov::hint::ExecutionMode::PERFORMANCE) ? Precision::I8 : Precision::I16;
+    }
+
+    if (!inference_precision_value.empty()) {
+        inference_precision = FromString<ov::element::Type>(inference_precision_value, ov::element::undefined);
+        if ((inference_precision != ov::element::i8) && (inference_precision != ov::element::i16)) {
+            THROW_GNA_EXCEPTION << "Unsupported precision of GNA hardware, should be i16 or i8, but was: "
+                                << inference_precision_value;
+        }
+        gnaPrecision = (inference_precision == ov::element::i8) ? Precision::I8 : Precision::I16;
+        execution_mode = (inference_precision == ov::element::i8) ? ov::hint::ExecutionMode::PERFORMANCE
+                                                                  : ov::hint::ExecutionMode::ACCURACY;
+    }
+}
+
 void Config::UpdateFromMap(const std::map<std::string, std::string>& config) {
+    std::string inference_precision_value = "";
+    std::string execution_mode_value = "";
```

Comment:
why do you store them as string? We have enumerations from API 2.0

Reply:
Yes, I didn't have to. Now I'm calling FromString right in the if...else.